File: brd-android/kyc-flutter/lib/login/models/login_response.dart
import 'dart:convert';

LoginResponse loginResponseFromJson(String str) =>
    LoginResponse.fromJson(json.decode(str));

String loginResponseToJson(LoginResponse data) => json.encode(data.toJson());

class LoginResponse {
  LoginResponse({
    required this.result,
    this.error,
    required this.data,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) => LoginResponse(
    result: json['result'],
    error: json['error'],
    data: Data.fromJson(json['data']),
  );

  String result;
  dynamic error;
  Data data;

  Map<String, dynamic> toJson() => {
        'result': result,
        'error': error,
        'data': data.toJson(),
      };
}

class Data {
  Data({
    required this.sessionKey,
    required this.needMfaToken,
    required this.authappEnabled,
    required this.smsEnabled,
    required this.emailEnabled,
  });

  factory Data.fromJson(Map<String, dynamic> json) => Data(
    sessionKey: json['sessionKey'],
    needMfaToken: json['needMfaToken'],
    authappEnabled: json['authapp_enabled'],
    smsEnabled: json['sms_enabled'],
    emailEnabled: json['email_enabled'],
  );

  String sessionKey;
  bool needMfaToken;
  int authappEnabled;
  int smsEnabled;
  int emailEnabled;

  Map<String, dynamic> toJson() => {
        'sessionKey': sessionKey,
        'needMfaToken': needMfaToken,
        'authapp_enabled': authappEnabled,
        'sms_enabled': smsEnabled,
        'email_enabled': emailEnabled,
      };
}

File: brd-android/signup/src/main/java/com/fabriik/signup/data/responses/LoginResponse.kt
package com.fabriik.signup.data.responses

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class LoginResponse(
    @Json(name = "sessionKey")
    val sessionKey: String,

    @Json(name = "is_confirmed")
    val isConfirmed: Boolean,

    @Json(name = "needMfaToken")
    val needMfaToken: Boolean,

    @Json(name = "authapp_enabled")
    val authAppEnabled: Int,

    @Json(name = "sms_enabled")
    val smsEnabled: Int,

    @Json(name = "email_enabled")
    val emailEnabled: Int,
)